portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,122 @@
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+const extraContent = document.querySelector("#extra-content");
+function extrasAreVisible(visible) {
+    if (visible) {
+        extraContent.style.display = "block"; 
+        extraContent.style.opacity = "1";  
+        //TODO: fix below line, which doesn't transition the opacity
+        extraContent.classList.add("change-extra-content-opacity"); 
+
+    } else {
+        extraContent.style.display = "none"; 
+    }
+}
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+const root = document.documentElement;
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+const secretText = document.querySelector("#secret-text"); 
+const mePic = document.querySelector("#me-pic"); 
+const tomiePic = document.querySelector("#tomie-pic"); 
+const mainHeading = document.querySelector("#main-heading"); 
+
+//fade in main text
+const mainText = document.querySelector(".main-text");
+window.addEventListener('load', e => {
+    mainText.classList.add("fade-text-in"); 
+    extrasAreVisible(false); 
+})
+
+/**
+ * Keeps track of mouse and updates our special, circle pointer to
+ * follow cursor movement. 
+ */
+const circleAroundMouse = document.querySelector(".cursor");
+window.addEventListener('mousemove', e => {
+    circleAroundMouse.style.top = e.pageY + "px"; 
+    circleAroundMouse.style.left = e.pageX + "px"; 
+})
+
+const secretMessages = ["...", "....", "Would you like to see something fun?"]; 
+let messageIndex = 0; 
+
+/**
+ * Animates the mouse as it interacts with section headers. 
+ */
+const sections = document.querySelectorAll(".sections li");
+sections.forEach(section => {
+    section.addEventListener("mouseover", e => {
+        circleAroundMouse.classList.add("grow-cursor");
+        if (eerieModeOn) { section.classList.remove("eerie-text"); } 
+        section.classList.add("color-text");  
+    }); 
+    section.addEventListener("mouseleave", e => {
+        circleAroundMouse.classList.remove("grow-cursor");
+        if (eerieModeOn) { section.classList.add("eerie-text"); } 
+        section.classList.remove("color-text"); 
+    }); 
+    section.addEventListener("click", e => {
+        if (section.id === "secret-text") {
+            mainText.textContent = secretMessages[messageIndex]; 
+            if (messageIndex < secretMessages.length) {
+                messageIndex += 1; 
+            } else {
+                activateEerieMode(); 
+            }
+        } else {
+            mainText.textContent = section.dataset.message; 
+        }
+    })
+});
+
+document.addEventListener('keydown', function(event) {
+    if(event.keyCode == 39) {
+        activateEerieMode(); 
+    }
+});
+
+/**
+ * Changes the theme of the page 
+ */
+function activateEerieMode() {
```

Comment:
I'll suggest moving this function closer to the top before the `sections.forEach`.
While reveiwing that section, I was confused about the `eerie-text` class (cos I couldn't find it in the html) until I found it here.